### PR TITLE
Adding context flag function and updating all components to use it

### PR DESCRIPTION
**Pull request by mohammedzee1000**
## What is the purpose of this change? What does it change?

This PR aims to add context flag to all commands that should have them

## How to test changes?

```
$ odo create nodejs --context ~/nodejs-ex/helm
 ✓  Checking component
 ✓  Checking component version
Please use `odo push` command to create the component with source deployed
```
This should work for most commands with exceptions as listed

### CODE REVIEW

File: pkg/odo/cli/cli.go [hunk 1]
```diff
@@ -150,6 +151,22 @@ func reconfigureCmdWithSubcmd(cmd *cobra.Command) {
...
+	}
+	if len(cmd.Commands()) == 0 {
+		return
+	}
+	for _, ei := range exceptionList {
+		if cmd.Name() == ei {
+			return
+		}
+	}
+	cmd.Flags().String("context", "", "define the context for this command")
```
Comment (mohammedzee1000) [inline pkg/odo/cli/cli.go]: Need to find a better message. This is to show it works

**Comment by girishramnani** [comment]
How would a command use this context, lets say `odo config --context` needs to access the context value?

**Comment by mohammedzee1000** [comment]
Been playing around. Found better approach PTAL

**Comment by surajnarwade** [comment]
@mohammedzee1000 , please remove WIP from title if it's ready for review :smile_cat:

**Comment by mohammedzee1000** [comment]
Just need to implement delete

**Comment by kadel** [comment]
> Just need to implement delete

are you going to do it here, or in another PR?

**Comment by girishramnani** [comment]
/hold
waiting for `context` in `delete`

**Pull request merged**